Bottom: 713a4922f071df56155e8cf28e85b231053844e6
Top:    9f16380974984f5c073d95c6e19f432036a0bd9a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 19:54:09 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 460ad55..17fb3f3 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2826,12 +2826,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
